components/dialog/RotateEncryptionKeyDialog.vue

```diff
@@ -26,25 +28,31 @@ export default {
   async fetch() {
     const c = this.cluster;
 
-    if ( c.isRke1 && this.$store.getters['isRancher'] ) {
-      let etcdBackups = await this.$store.dispatch('rancher/findAll', { type: NORMAN.ETCD_BACKUP });
+    let etcdBackups = [];
 
+    if ( c.isRke1 && this.$store.getters['isRancher'] ) {
+      etcdBackups = await this.$store.dispatch('rancher/findAll', { type: NORMAN.ETCD_BACKUP });
       etcdBackups = etcdBackups.filter(backup => backup.clusterId === c.metadata.name);
+    }
 
-      const backups = sortBy(etcdBackups, ['created']).reverse();
-      const dateFormat = escapeHtml(this.$store.getters['prefs/get'](DATE_FORMAT));
-      const timeFormat = escapeHtml( this.$store.getters['prefs/get'](TIME_FORMAT));
+    if (c.isRke2 && this.$store.getters['isRancher']) {
```

Comment:
I'm rewriting it so that `else if` will not required. It will just not execute the code for RKE2 if it's RKE1 and vice versa:
```
const getEtcdBackups = () => {
      if ( c.isRke1) {
        let etcdBackups = await this.$store.dispatch('rancher/findAll', { type: NORMAN.ETCD_BACKUP });
        return etcdBackups.filter(backup => backup.clusterId === c.metadata.name);
      }

      if (c.isRke2 && this.$store.getters['isRancher']) {
        let etcdBackups = await this.$store.dispatch('management/findAll', { type: SNAPSHOT });
        return etcdBackups.filter(backup => backup.clusterId === c.id);
      }
      return [];
    }
```